src/openrct2/world/TileInspector.cpp

```diff
@@ -896,59 +864,28 @@ int32_t tile_inspector_track_set_chain(
         trackBlock += trackElement->AsTrack()->GetSequenceIndex();
 
         uint8_t originDirection = trackElement->GetDirection();
-        switch (originDirection)
-        {
-            case 0:
-                originX -= trackBlock->x;
-                originY -= trackBlock->y;
-                break;
-            case 1:
-                originX -= trackBlock->y;
-                originY += trackBlock->x;
-                break;
-            case 2:
-                originX += trackBlock->x;
-                originY += trackBlock->y;
-                break;
-            case 3:
-                originX += trackBlock->y;
-                originY -= trackBlock->x;
-                break;
-        }
+        CoordsXY offsets = { trackBlock->x, trackBlock->y };
+        CoordsXY coords = { originX, originY };
+        coords += offsets.Rotate(direction_reverse(originDirection));
 
+        originX = (int16_t)coords.x;
+        originY = (int16_t)coords.y;
         originZ -= trackBlock->z;
 
         trackBlock = get_track_def_from_ride(ride, type);
         for (; trackBlock->index != 255; trackBlock++)
         {
-            int16_t elemX = originX, elemY = originY, elemZ = originZ;
-
-            switch (originDirection)
-            {
-                case 0:
-                    elemX += trackBlock->x;
-                    elemY += trackBlock->y;
-                    break;
-                case 1:
-                    elemX += trackBlock->y;
-                    elemY -= trackBlock->x;
-                    break;
-                case 2:
-                    elemX -= trackBlock->x;
-                    elemY -= trackBlock->y;
-                    break;
-                case 3:
-                    elemX -= trackBlock->y;
-                    elemY += trackBlock->x;
-                    break;
-            }
-
+            CoordsXY elem = { originX, originY };
+            int16_t elemZ = originZ;
+            offsets.x = trackBlock->x;
+            offsets.y = trackBlock->y;
```

Comment:
I'm not really a fan of reusing this variable for a different value we should really create a new variable.

Reply:
Did you want me to resolve this issue in this PR, or a future refactor?

Reply:
Just leave it.